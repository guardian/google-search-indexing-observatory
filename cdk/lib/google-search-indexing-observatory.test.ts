import { App } from "aws-cdk-lib";
import { Template } from "aws-cdk-lib/assertions";
import { GoogleSearchIndexingObservatory } from "./google-search-indexing-observatory";

describe("The GoogleSearchIndexingObservatory stack", () => {
  it("matches the snapshot", () => {
    const app = new App();
    const stack = new GoogleSearchIndexingObservatory(app, "GoogleSearchIndexingObservatory", {
      stack: "ophan",
      stage: "TEST",
      withBackup: true,
    });
    const template = Template.fromStack(stack);
    expect(template.toJSON()).toMatchSnapshot();
  });
});
