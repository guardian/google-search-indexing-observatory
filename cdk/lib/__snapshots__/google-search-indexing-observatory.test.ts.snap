// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The GoogleSearchIndexingObservatory stack matches the snapshot 1`] = `
Object {
  "Metadata": Object {
    "gu:cdk:constructs": Array [
      "GuDistributionBucketParameter",
      "GuScheduledLambda",
    ],
    "gu:cdk:version": "TEST",
  },
  "Parameters": Object {
    "DistributionBucketName": Object {
      "Default": "/account/services/artifact.bucket",
      "Description": "SSM parameter containing the S3 bucket name holding distribution artifacts",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": Object {
    "TableCD117FA1": Object {
      "DeletionPolicy": "Retain",
      "Properties": Object {
        "AttributeDefinitions": Array [
          Object {
            "AttributeName": "uri",
            "AttributeType": "S",
          },
        ],
        "KeySchema": Array [
          Object {
            "AttributeName": "uri",
            "KeyType": "HASH",
          },
        ],
        "ProvisionedThroughput": Object {
          "ReadCapacityUnits": 5,
          "WriteCapacityUnits": 5,
        },
        "Tags": Array [
          Object {
            "Key": "devx-backup-enabled",
            "Value": "true",
          },
          Object {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          Object {
            "Key": "gu:repo",
            "Value": "guardian/google-search-indexing-observatory",
          },
          Object {
            "Key": "Stack",
            "Value": "ophan",
          },
          Object {
            "Key": "Stage",
            "Value": "TEST",
          },
        ],
      },
      "Type": "AWS::DynamoDB::Table",
      "UpdateReplacePolicy": "Retain",
    },
    "scheduledLambda8A84450D": Object {
      "DependsOn": Array [
        "scheduledLambdaServiceRoleDefaultPolicyDE6B5E9F",
        "scheduledLambdaServiceRoleB98DFEFD",
      ],
      "Properties": Object {
        "Code": Object {
          "S3Bucket": Object {
            "Ref": "DistributionBucketName",
          },
          "S3Key": "ophan/TEST/google-search-indexing-observatory/google-search-indexing-observatory.jar",
        },
        "Environment": Object {
          "Variables": Object {
            "APP": "google-search-indexing-observatory",
            "STACK": "ophan",
            "STAGE": "TEST",
          },
        },
        "Handler": "ophan.google.indexing.observatory.Lambda::handler",
        "MemorySize": 1024,
        "Role": Object {
          "Fn::GetAtt": Array [
            "scheduledLambdaServiceRoleB98DFEFD",
            "Arn",
          ],
        },
        "Runtime": "java11",
        "Tags": Array [
          Object {
            "Key": "App",
            "Value": "google-search-indexing-observatory",
          },
          Object {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          Object {
            "Key": "gu:repo",
            "Value": "guardian/google-search-indexing-observatory",
          },
          Object {
            "Key": "Stack",
            "Value": "ophan",
          },
          Object {
            "Key": "Stage",
            "Value": "TEST",
          },
        ],
        "Timeout": 30,
      },
      "Type": "AWS::Lambda::Function",
    },
    "scheduledLambdaServiceRoleB98DFEFD": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
        "Tags": Array [
          Object {
            "Key": "App",
            "Value": "google-search-indexing-observatory",
          },
          Object {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          Object {
            "Key": "gu:repo",
            "Value": "guardian/google-search-indexing-observatory",
          },
          Object {
            "Key": "Stack",
            "Value": "ophan",
          },
          Object {
            "Key": "Stage",
            "Value": "TEST",
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "scheduledLambdaServiceRoleDefaultPolicyDE6B5E9F": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
              ],
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:",
                      Object {
                        "Ref": "AWS::Partition",
                      },
                      ":s3:::",
                      Object {
                        "Ref": "DistributionBucketName",
                      },
                    ],
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:",
                      Object {
                        "Ref": "AWS::Partition",
                      },
                      ":s3:::",
                      Object {
                        "Ref": "DistributionBucketName",
                      },
                      "/ophan/TEST/google-search-indexing-observatory/google-search-indexing-observatory.jar",
                    ],
                  ],
                },
              ],
            },
            Object {
              "Action": "ssm:GetParametersByPath",
              "Effect": "Allow",
              "Resource": Object {
                "Fn::Join": Array [
                  "",
                  Array [
                    "arn:aws:ssm:",
                    Object {
                      "Ref": "AWS::Region",
                    },
                    ":",
                    Object {
                      "Ref": "AWS::AccountId",
                    },
                    ":parameter/TEST/ophan/google-search-indexing-observatory",
                  ],
                ],
              },
            },
            Object {
              "Action": Array [
                "ssm:GetParameters",
                "ssm:GetParameter",
              ],
              "Effect": "Allow",
              "Resource": Object {
                "Fn::Join": Array [
                  "",
                  Array [
                    "arn:aws:ssm:",
                    Object {
                      "Ref": "AWS::Region",
                    },
                    ":",
                    Object {
                      "Ref": "AWS::AccountId",
                    },
                    ":parameter/TEST/ophan/google-search-indexing-observatory/*",
                  ],
                ],
              },
            },
            Object {
              "Action": Array [
                "dynamodb:BatchGetItem",
                "dynamodb:GetRecords",
                "dynamodb:GetShardIterator",
                "dynamodb:Query",
                "dynamodb:GetItem",
                "dynamodb:Scan",
                "dynamodb:ConditionCheckItem",
                "dynamodb:BatchWriteItem",
                "dynamodb:PutItem",
                "dynamodb:UpdateItem",
                "dynamodb:DeleteItem",
                "dynamodb:DescribeTable",
              ],
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::GetAtt": Array [
                    "TableCD117FA1",
                    "Arn",
                  ],
                },
                Object {
                  "Ref": "AWS::NoValue",
                },
              ],
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "scheduledLambdaServiceRoleDefaultPolicyDE6B5E9F",
        "Roles": Array [
          Object {
            "Ref": "scheduledLambdaServiceRoleB98DFEFD",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "scheduledLambdascheduledLambdarate1minute07D1D4D61": Object {
      "Properties": Object {
        "ScheduleExpression": "rate(1 minute)",
        "State": "ENABLED",
        "Targets": Array [
          Object {
            "Arn": Object {
              "Fn::GetAtt": Array [
                "scheduledLambda8A84450D",
                "Arn",
              ],
            },
            "Id": "Target0",
          },
        ],
      },
      "Type": "AWS::Events::Rule",
    },
    "scheduledLambdascheduledLambdarate1minute0AllowEventRuleGoogleSearchIndexingObservatoryscheduledLambda2A2AC158D4E5E85C": Object {
      "Properties": Object {
        "Action": "lambda:InvokeFunction",
        "FunctionName": Object {
          "Fn::GetAtt": Array [
            "scheduledLambda8A84450D",
            "Arn",
          ],
        },
        "Principal": "events.amazonaws.com",
        "SourceArn": Object {
          "Fn::GetAtt": Array [
            "scheduledLambdascheduledLambdarate1minute07D1D4D61",
            "Arn",
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
  },
}
`;
